gammapy/time/lightcurve.py

```diff
@@ -112,3 +116,249 @@ def compute_fvar(self):
         fvar_err = sigxserr / (2 * fvar)
 
         return fvar, fvar_err
+
+
+class LightCurveFactory(object):
+    """
+    Class producing light curve
+
+    Parameters
+    ----------
+    spectral_extraction: `~gammapy.spectrum.SpectrumExtraction`
+       class storing statistics, IRF and event lists
+    """
+
+    def __init__(self, spectral_extraction):
+        self.obs_list = spectral_extraction.obs
+        self.obs_spec = spectral_extraction.observations
+        self.off_evt_list = self._get_off_evt_list(spectral_extraction)
+        self.on_evt_list = self._get_on_evt_list(spectral_extraction)
+
+        
+    def _get_off_evt_list(self, spectral_extraction):
+        """
+        Returns list of OFF events for each observations
+        """
+        off_evt_list = []
+        for bg in spectral_extraction.background:
+            off_evt_list.append(bg.off_events)
+        return off_evt_list
+
+    
+    def _get_on_evt_list(self, spectral_extraction):
+        """
+        Returns list of OFF events for each observations
+        """
+        on_region = spectral_extraction.target.on_region
+        on_evt_list = []
+        for obs in spectral_extraction.obs:
+            idx = on_region.contains(obs.events.radec)
+            on_evt_list.append(obs.events.select_row_subset(idx))
+            
+        return on_evt_list
+    
+            
+    def _get_time_intervals(self, t_start, t_stop, time_binning_type):
```

Comment:
> So my suggestion would be to not take t_start, t_stop, t_binning_type in the lightcurve method,
> and to instead somehow expose a function or method to compute time binnings.

Done, I added a `time_intervals` argument.

> One question is which things in this class to take as input in __init__ and which in the lightcurve or "run" method.
> Maybe it doesn't matter much?

 I keep the `spectral_model` and `energy_range` arguments since they are needed to compute the light curve. My goal was to be able to compute different time-binned light curves with the light_curve function for a given spectral extraction. With this we are able to easily compute light curves with different energy threshold (e.g. >100 GeV, >1 TeV), for a given model. Maybe someone might be interested to also test different spectral model (maybe less frequent that the energy threshold use case).

> Another question is how to represent the time binning.
> In your current method you don't have a Returns section in the docstring, but from the code it looks like it's a Python list of scalar Time objects.
> That's not so bad, but maybe using
> http://docs.gammapy.org/en/latest/api/gammapy.data.GTI.html
> for that purpose would be better (i.e. an Astropy table o TSTART / TSTOP values, exposed as Time attributes)?
> Then for that we have I/O already and we need to develop methods anyways to e.g. merge and clip GTIs from runs, so in the future there could be conveniences to work with GTIs and that could be used for people preparing a lightcurve time binning?

For now I put aside this question, i stick with astropy.time.core.Time objects.

> The last question would be where the code to compute the time binning using the method you have here should live. I'm not sure. It could live on the LightCurve class, on the GTI class, on the ObservationList class, or in a new function or class in this file where the LightCurveFactory is.

Same here, I removed the `_get_time_intervals` function. We'll do something fancy later.